src/scifem/eval.py

```diff
@@ -0,0 +1,74 @@
+from mpi4py import MPI
+import dolfinx
+import numpy as np
+import numpy.typing as npt
+
+
+def evaluate_function(
```

Comment:
Should we just call it `eval` or `evaluate`, so that it can be used for `Expression` later without any API change?

Reply:
We can start with `evaluate_function` and then reconsider the name when implementing this for expression later